code-challenges.js

```diff
@@ -41,35 +67,81 @@ var vowelTester1 = "learn"
+  //Find the first index of y after index 0, store in the vowelFirstIndices array
+  string.split("").map((value, index) => {if (string[index] === "y" && index !== 0) {vowelFirstIndices.push(index)}})
+  //Return the lowest number from the vowelFirstIndices array
+  return Math.min(...vowelFirstIndices)
+}
+
+console.log(firstVowel(vowelTester1)); //expect: 1
+console.log(firstVowel(vowelTester2)); //expect: 3
+console.log(firstVowel(vowelTester3)); //expect: 2
+console.log(firstVowel(vowelTester4)); //expect: 1
 
 // --------------------5) Create a function that takes three arguments - two numbers and a mathematical operation (+, -, *, /) and performs the given calculation. If the input tries to divide by 0, return: "Can't divide by 0!"
 
-
-
-
+//Write a function that takes in two numbers and one mathematical operation
+const calculator = (number1, operator, number2) => {
+  //Check if the user is trying to divide by zero
+  if (operator === "/" && number2 === 0) {
+    //If so, return an error message as specified
+    return "Can't divide by 0!"
+  } else {
+    //Otherwise, use the evaluate method to process the arguments in order
+    return eval(`${number1} ${operator} ${number2}`)
+  }
+}
 
 // Uncomment and use the following console logs to test your function
-// console.log(calculator(3, "*", 9))
+console.log(calculator(3, "*", 9))
 // Expected output: 27
 
-// console.log(calculator(16, "+", 3))
+console.log(calculator(16, "+", 3))
 // Expected output: 19
 
-// console.log(calculator(89, "/", 0))
+console.log(calculator(89, "/", 0))
 // Expected output: "Can't divide by 0!"
 
-
-
 // --------------------6) Create a function that takes in the following variable and returns only the websites that end in ".io".
 
 var websites = ["codepen.io", "codecademy.com", "coursera.org", "codepen.io", "udemy.com", "pluralsight.com", "udacity.com", "sitepoint.com"]
 // Expected output: "codepen.io" "codepen.io"
 
+//Create a function that takes in an array of strings
+const ioFinder = (array) => {
+  //Use the slice method within the filter method to return a new array of strings where the last three characters are ".io"
+  return array.filter(value => value.slice(-3) === ".io")
+}
 
-
+console.log(ioFinder(websites)); //expect [codepen.io, codepen.io]
 
 // --------------------6) STRETCH: Write a function that takes a number as an argument and uses a WHILE loop to count up to that number from 0.
+
+//Create a function that takes in a number
+const countUp = (number) => {
+  //Initialize a variable to start the count from 0
+  var count = 0
+  //Use a while loop to iterate a total of number + 1 times
+  while (count < number + 1) {
+    //Display the current count in the console
+    console.log(count);
+    //Increment the count
+    count++
+  }
+}
+
+countUp(3) //expect 0, 1, 2, 3
```

Comment:
Wrote this last night since I misunderstood the problem first time around:

const countUp = (number) => {
    //Initialize a variable to start the count from 0
    var count = 0
    //Initialize the string that will store all numbers counting up
    let answerCount = ""
    //Use a while loop to iterate a total of number + 1 times
    while (count < number + 1) {
      //Add the current count number as a string to answerCount, then add a comma with a trailing space
      answerCount += count.toString() + ", "
      //Increment the count
      count++
    }
    //Return answerCount after cutting off the final comma and trailing space
    return answerCount.slice(0, -2)
  }
  
  console.log(countUp(3)) //expect 0, 1, 2, 3